Module 5, Block 1 : Blog App - processing data in models

aceDavon opened the pull request:
## In this PR, I have implemented the following project requirements: :writing_hand: 

- [ ] Set up associations between models.
- [ ] Write SQL queries with Active Record.

aceDavon committed:
Add database models to project

Create database models for posts, users, likes and comments

aceDavon committed:
Add relationships to models

Add database relationships to models and implement counter functions

AmaduKamara commented on app/models/comment.rb:
```diff
@@ -0,0 +1,9 @@
+  after_save :update_comments_counter
+
+  belongs_to :post
+
```
- [ ] According to the ERD given, a comment has to belong to an author and post, but your association only considers the post. Kindly add an association for a comment to belong to an author who wrote the comment 👍

AmaduKamara commented on app/models/like.rb:
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  after_save :update_likes_counter
+  belongs_to :post
+
```
- [ ] A like is supposed to belong to an author. From the ERD, a like is associated with a person, which can be an author. Kindly consider that 👍

AmaduKamara reviewed (changes requested):
# Status: Required Changes ♻️♻️♻️ 

Hi @aceDavon,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍 
✅ Linter checks passed 👍 
✅ README is okay 👍 
✅ Good commit messages 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

aceDavon committed:
Add user association to like and comment

implement the user association on the like and comment model to match the
ERD diagram as described by the project requirement

prolajumokeoni reviewed (approved):
Hi @aceDavon 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
